airflow/models/dagrun.py

```diff
@@ -859,6 +859,8 @@ def get_previous_scheduled_dagrun(
         :param session: SQLAlchemy ORM Session
         """
         dag_run = session.get(DagRun, dag_run_id)
+        if not dag_run.logical_date:
+            return None
```

Comment:
In what case it make sense that logical_date is empty?

Reply:
That happened in AIP-83 amendment

Reply:
ah right